src/service/context/extensions.rs

```diff
@@ -98,6 +98,13 @@ impl Extensions {
         });
         ext
     }
+    /// Checks if a value of type T exists in the extensions map
+    pub fn contains<T: 'static + Send + Sync>(&self) -> bool {
```

Comment:
Don't think you need this function. Contains serves not much use in actual usage from non-context code.

I think better that you implement `get_or_insert_default`, `get_or_insert` and `get_or_insert_with` directly in `Extensions` and then expose these as-is from the `Context`.

I do realize I am asking perhaps a lot from you so do let me know at any point if you need me to take over.

And of course if you do not agree, do let me know.

Reply:
I want to clarify what is the difference between getting_or_insert and get_or_insert with ?

Reply:
- `get_or_insert`: insert a value `t: T` which is pre-created (so the function takes a `t: T`)
- `get_or_insert_with`: takes a closure/function that creates a value `T`, so is lazy
- `get_or_insert_default`: also lazy, but instead of needing a closure it can just call `T::default()`

An analog example of this concept can be found in the std lib: <https://doc.rust-lang.org/std/option/#modifying-an-option-in-place>